ios/CoinbaseWalletSDK/CoinbaseWalletSDK.swift

```
@@ -167,13 +150,18 @@ public final class CoinbaseWalletSDK {
     /// - Parameter url: deep link url
     /// - Returns: `false` if the input was not response message type, `true` if SDK handled the input, or throws error if it failed to decode response.
     @discardableResult
-    public func handleResponse(_ url: URL) throws -> Bool {
-        guard isWalletSegueMessage(url) else {
+    static public func handleResponse(_ url: URL) throws -> Bool {
+        let encryptedResponse: EncryptedResponseMessage = try MessageConverter.decodeWithoutDecryption(url)
+        guard let instance = instances[URL(string: encryptedResponse.callbackUrl ?? "")!] else {
```

why not checking this after `isWalletSegueMessage`?
also, it would make more sense to throw an error for this case, instead of returning just boolean

---

updated to move isWalletSegueMessage prior to instance retrieval